### PR TITLE
UIDATIMP-1182: Change UI flow for uploaded files deleted before import by users

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

* Instead of waiting, change the UI behavior so that an "Are you sure" modal is displayed when a user clicks the trashcan. If yes, delete the file immediately. If no, leave the file on the Choose jobs page.

## Approach

* Added `<ConfirmationModal>` to confirm or cancel deleting file
* Replaced **has been deleted** text with **will be deleted** when click on the trash icon
*  If an uploaded file is deleted, and it is the last uploaded file, then redirect user to Data Import landing page, using `history.push`

## Refs

[UIDATIMP-1182](https://issues.folio.org/browse/UIDATIMP-1182)

## Screenshots

https://user-images.githubusercontent.com/85172747/181839740-f81527a3-4c42-4879-9280-32c6558bc503.mp4

### CODE REVIEW

**Review by Javokhir12 — APPROVED** [review]

File: src/components/UploadingJobsDisplay/UploadingJobsDisplay.js [hunk 1]
```diff
@@ -415,13 +408,7 @@ export class UploadingJobsDisplay extends Component {
   });
 
   handleDeleteSuccessfullyUploadedFile = (fileKey, fileStatus) => {
```
Comment (mariia-aloshyna) [inline src/components/UploadingJobsDisplay/UploadingJobsDisplay.js:410]: Do we need this additional wrapper method if it just calls deleteFileAPI?

File: src/components/UploadingJobsDisplay/UploadingJobsDisplay.js [hunk 2]
```diff
@@ -480,6 +467,17 @@ export class UploadingJobsDisplay extends Component {
     this.setState({ renderLeaveModal: false });
   };
 
+  openCancelUploadModal = (fileKey) => {
```
Comment (mariia-aloshyna) [inline src/components/UploadingJobsDisplay/UploadingJobsDisplay.js:470]:
```suggestion
  openCancelUploadModal = fileKey => {
```

File: src/components/UploadingJobsDisplay/components/FileItem/FileItem.js [hunk 3]
```diff
@@ -13,8 +13,8 @@ import css from './FileItem.css';
 export const FileItem = memo(({
   uiKey,
   status = FILE_STATUSES.UPLOADING,
+  onCancel,
```
Comment (mariia-aloshyna) [inline src/components/UploadingJobsDisplay/components/FileItem/FileItem.js:16]:
```suggestion
  onCancelImport,
```

File: src/components/UploadingJobsDisplay/tests/UploadingJobsDisplay.test.js [hunk 4]
```diff
@@ -214,8 +248,25 @@ describe('<UploadingjobsDisplay>', () => {
         uploadDefinition: { fileDefinitions: [{ status: FILE_STATUSES.UPLOADING }] },
       });
 
+      window.dispatchEvent(new Event('beforeunload'));
+
+      // history.push({ location: { pathname: '/settings' } });
```
Comment (mariia-aloshyna) [inline src/components/UploadingJobsDisplay/tests/UploadingJobsDisplay.test.js:253]: Remove this, please

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**